pages/[walletAddress]/proposals.tsx

```diff
@@ -0,0 +1,308 @@
+import { tzip16 } from "@taquito/tzip16";
+import { validateContractAddress } from "@taquito/utils";
+import {
+  useContext,
+  useEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  useState,
+} from "react";
+import ProposalCard from "../../components/ProposalCard";
+import Spinner from "../../components/Spinner";
+import Meta from "../../components/meta";
+import Modal from "../../components/modal";
+import ProposalSignForm from "../../components/proposalSignForm";
+import fetchVersion from "../../context/metadata";
+import { getProposals } from "../../context/proposals";
+import { AppStateContext } from "../../context/state";
+import { proposal, version } from "../../types/display";
+import { canExecute, canReject } from "../../utils/proposals";
+import useIsOwner from "../../utils/useIsOwner";
+import useWalletTokens from "../../utils/useWalletTokens";
+import { getProposalsId, signers, toProposal } from "../../versioned/apis";
+
+const emptyProps: [number, { og: any; ui: proposal }][] = [];
+
+type proposals = [number, { og: any; ui: proposal }][];
+type openModal = {
+  state: number;
+  proposal: [boolean | undefined, number];
+};
+type state = {
+  isLoading: boolean;
+  isInvalid: boolean;
+  proposals: proposals;
+  openModal: openModal;
+  currentAddress: string | null;
+};
+
+type action =
+  | { type: "setLoading"; payload: boolean }
+  | { type: "setInvalid"; payload: boolean }
+  | { type: "setProposals"; payload: { proposals: proposals; address: string } }
+  | { type: "setOpenModal"; payload: openModal }
+  | { type: "setOpenModalState"; payload: number };
+
+const reducer = (state: state, action: action): state => {
+  switch (action.type) {
+    case "setLoading":
+      return { ...state, isLoading: action.payload };
+    case "setInvalid":
+      return { ...state, isInvalid: action.payload };
+    case "setProposals":
+      return {
+        ...state,
+        proposals: action.payload.proposals,
+        currentAddress: action.payload.address,
+        isLoading: false,
+      };
+    case "setOpenModal":
+      return { ...state, openModal: action.payload };
+    case "setOpenModalState":
+      return {
+        ...state,
+        openModal: {
+          ...state.openModal,
+          state: action.payload,
+        },
+      };
+  }
+};
+const Proposals = () => {
+  const globalState = useContext(AppStateContext)!;
+  const isOwner = useIsOwner();
+  const walletTokens = useWalletTokens();
+
+  const [state, dispatch] = useReducer<typeof reducer>(reducer, {
+    isLoading: true,
+    isInvalid: false,
+    currentAddress: globalState.currentContract,
+    proposals: [],
+    openModal: {
+      state: 0,
+      proposal: [undefined, 0],
+    },
+  });
+
+  const previousRefresherRef = useRef(-1);
+  const [refresher, setRefresher] = useState(0);
+
+  useEffect(() => {
```

Comment:
currently, if you have an imported wallet and you use `/<another wallet>/proposals` to visit `another wallet`, the page will show "an imported wallet" first after few second it will generate `another wallet`. suggestion as following:
```
const router = useRouter();
```

```suggestion
  useEffect(() => {
     if (!router.isReady) return;
```

Reply:
fixed, as discussion offline